### ISSUE TITLE
Recent commit broke the gpt usage (for me)

### ISSUE THREAD

**Issue by schnika** [kharvd/gpt-cli#29]
Current behaviour (https://github.com/kharvd/gpt-cli/commit/46247d3b181fda1c307fccf45793fdc41740083a)

```
~/p/g/gpt-cli ❯❯❯ ./gpt.py dev                                                 main
Traceback (most recent call last):
  File "./gpt.py", line 10, in <module>
    from gptcli.assistant import (
  File "/Users/patricknueser/projects/gpt-cli/gpt-cli/gptcli/assistant.py", line 8, in <module>
    from gptcli.llama import LLaMACompletionProvider
  File "/Users/patricknueser/projects/gpt-cli/gpt-cli/gptcli/llama.py", line 12, in <module>
    LLAMA_MODELS: Optional[dict[str, Path]] = None
TypeError: 'type' object is not subscriptable
```

Expected behaviour (7752b05b2245136746c355e9586cf846790b268c)
~/p/g/gpt-cli ❯❯❯ ./gpt.py dev                                           ✘ 1 main
Hi! I'm here to help. Type q or Ctrl-D to exit, c or Ctrl-C to clear the
conversation, r or Ctrl-R to re-generate the last response. To enter multi-line
mode, enter a backslash \ followed by a new line. Exit the multi-line mode by
pressing ESC and then Enter (Meta+Enter).
>

**Comment by kharvd** [comment]
Hmm, I can't reproduce on a clean installation. What is your python version? GPT-4 thinks you're using an outdated version. Could you try the suggested fix below as well?

```
The error is caused by using dict[str, Path] with an older Python version that doesn't support it. Upgrade
to Python 3.9 or later, or use typing.Dict:

 from typing import Dict, Optional
 LLAMA_MODELS: Optional[Dict[str, Path]] = None
```

**Comment by sthiffea** [comment]
I'm having same issue. Python 3.8.10 on Ubuntu 20.04 under WSL2.
Tried clean install. Same issue.
```(venv) sthiffea@win11desktop:~/gpt-cli$ ./gpt.py
Traceback (most recent call last):
  File "./gpt.py", line 8, in <module>
    import google.generativeai as genai
  File "/home/sthiffea/gpt-cli/venv/lib/python3.8/site-packages/google/generativeai/__init__.py", line 59, in <module>
    from google.generativeai import types
  File "/home/sthiffea/gpt-cli/venv/lib/python3.8/site-packages/google/generativeai/types/__init__.py", line 19, in <module>
    from google.generativeai.types.text_types import *
  File "/home/sthiffea/gpt-cli/venv/lib/python3.8/site-packages/google/generativeai/types/text_types.py", line 34, in <module>
    class Completion(abc.ABC):
  File "/home/sthiffea/gpt-cli/venv/lib/python3.8/site-packages/google/generativeai/types/text_types.py", line 49, in Completion
    filters: Optional[list[safety_types.ContentFilterDict]]```

**Comment by dltn** [comment]
I've also hit this on a clean install Mac. The issue occurs for any Python <3.9 due to the typing error in upstream `google-generativeai`, discussed in https://github.com/google/generative-ai-python/issues/13 and fixed in https://github.com/google/generative-ai-python/pull/16.

In addition to fixing the type issue, `google-generativeai` has also dropped support for any Python <3.9 in `0.1.0rc3`/`0.1.0`, which means it isn't as simple as bumping the package:

```
% diff --git a/requirements.txt b/requirements.txt
-google-generativeai==0.1.0rc2
+google-generativeai==0.1.0

% python3 -m pip install -r requirements.txt
ERROR: Ignored the following versions that require a different python version: 0.1.0 Requires-Python >=3.9; 0.1.0rc3 Requires-Python >=3.9
```

@kharvd: do you want to make `gpt-cli` Python 3.9+ only?

**Comment by dltn** [comment]
> @kharvd: do you want to make gpt-cli Python 3.9+ only?

If yes: https://github.com/kharvd/gpt-cli/pull/39

**Comment by kharvd** [comment]
Merged #39, Python >= 3.9 is now required, thanks @dltn
